google/cloud/bigquery/job.py

```diff
@@ -3160,7 +3160,12 @@ def _begin(self, client=None, retry=DEFAULT_RETRY, timeout=None):
             raise
 
     def result(
-        self, page_size=None, max_results=None, retry=DEFAULT_RETRY, timeout=None
+        self,
+        start_index=None,
+        page_size=None,
+        max_results=None,
+        retry=DEFAULT_RETRY,
+        timeout=None,
```

Comment:
Even though the new parameter's position seems logical and appealing, it can break any existing user code that passes the values as positional arguments. It would thus be more safe to add `start_index` at the end.